### PR TITLE
WIP Resize of change of layout attribute

### PR DESCRIPTION
**Pull request by SylvainCorlay**
@jasongrout @sccolbert comments?

Now I understand what is going on: 
- The box widget should not add children to the DOM manually but call attach.
- since onAfterAttach triggers jupyter-js-widgets'  `display` event, we should make it not trigger the displayed event.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
I see the event being triggered but the child widgets don't get it yet.

File: jupyter-js-widgets/src/widget.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ import {
     Widget
 } from 'phosphor/lib/ui/widget';
 
+import {
+  Message, sendMessage
```
Comment (sccolbert) [inline jupyter-js-widgets/src/widget.ts]: `Message` appears to be an unused import

**Comment by gnestor** [comment]
Is there anything I can do to help merge this one?

**Comment by jasongrout** [comment]
> Is there anything I can do to help merge this one?

Thanks for asking. What are your thoughts on the logic?

**Comment by gnestor** [comment]
@jasongrout I'm not deeply familiar with this project yet so I'm not sure about the logic. What I can do is test it!

@SylvainCorlay How exactly do I test the resize (or child layout change event) or a `DOMWidgetView`?

**Comment by jasongrout** [comment]
Closed in favor of #979.

**Pull request closed without merging**